app/views/layouts/blacklight/base.html.erb

```diff
@@ -28,7 +28,7 @@
   <main id="main-container" class="<%= container_classes %>" role="main" aria-label="<%= t('blacklight.main.aria.main_container') %>">
     <%= content_for(:container_header) %>
 
-    <%= render partial: 'shared/flash_msg', layout: 'shared/flash_messages' %>
+    <%= render partial: 'shared/flash_msg', layout: 'shared/flash_messages' unless params[:id] && current_page?(solr_document_path)%>
 
```

Comment:
there may be a better way to do this...basically, we don't want to render the flash messages in the layout if we are on detail page, as that is handled in the components (so they can be rendered in a different location instead)